railgun/manager/manager.go

```diff
@@ -333,6 +350,75 @@ func Run(ctx context.Context, c *Config) error {
 		return fmt.Errorf("unable to setup readyz: %w", err)
 	}
 
+	// if anonymous reports are enabled this helps provide Kong with insights about usage of the ingress controller
+	// which is non-sensitive and predominantly informs us of the controller and cluster versions in use.
+	// This data helps inform us what versions, features, e.t.c. end-users are actively using which helps to inform
+	// our prioritization of work and we appreciate when our end-users provide them, however if you do feel
+	// uncomfortable and would rather turn them off run the controller with the "--anonymous-reports false" flag.
+	reporterLogger := logrus.StandardLogger()
+	if c.AnonymousReports {
```

Comment:
I think that the main function of the ctrl manager is a good place for wiring/plumbing/dependency injection, not necessarily for logic.

Please put this big block in a function in a separate package, e.g. alongside `util.Reporter`.

To avoid polluting that logic with the logging concern, consider just returning an `error` from that function, and making a `logger.*` call just once, with the error value.

Reply:
Sounds good :+1: 20f8874285cb4105f138fc6fc9c3bf03f0466f5c

Reply:
Due to problems with import cycles, and some of the differences between 1x and 2x I opted to put this in `internal/mgrutils` let me know if that seems OK to you :+1:

Reply:
What about the other part:

> To avoid polluting that logic with the logging concern, consider just returning an `error` from that function, and making a `logger.*` call just once, with the error value.

Reply:
:+1: 2d090301c06482f76342f667736d6c609a977191